portlet-parent/wicketstuff-portlet/src/main/java/org/apache/wicket/portlet/WicketPortlet.java

```diff
@@ -652,7 +653,21 @@ protected String fixWicketUrl(final String requestUrl, final String url) {
 		if ((url != null) && (requestUrl != null) && (!ABSOLUTE_URI_PATTERN.matcher(url).matches())) {
 			try {
 				if (!requestUrl.startsWith("http")) {
-					return new URL(new URL("http:" + wicketFilterPath), url).toString().substring(5);
+					URL fixedUrl = new URL("http:" + url);
+					String query = fixedUrl.getQuery();
+					if (query != null) {
+						for (String queryParamValuePair : query.split("&")) {
+							String wuViewParam = "_wuview=";
```

Comment:
You are right, I will fix it as soon as possible.
Στις 11 Μαϊ 2016 10:39 μ.μ., ο χρήστης "Martin Grigorov" <
notifications@github.com> έγραψε:

> In
> portlet-parent/wicketstuff-portlet/src/main/java/org/apache/wicket/portlet/WicketPortlet.java
> <https://github.com/wicketstuff/core/pull/494#discussion_r62912504>:
>
> > @@ -652,7 +653,21 @@ protected String fixWicketUrl(final String requestUrl, final String url) {
> >  		if ((url != null) && (requestUrl != null) && (!ABSOLUTE_URI_PATTERN.matcher(url).matches())) {
> >  			try {
> >  				if (!requestUrl.startsWith("http")) {
> > -					return new URL(new URL("http:" + wicketFilterPath), url).toString().substring(5);
> > +					URL fixedUrl = new URL("http:" + url);
> > +					String query = fixedUrl.getQuery();
> > +					if (query != null) {
> > +						for (String queryParamValuePair : query.split("&")) {
> > +							String wuViewParam = "_wuview=";
>
> This could be out of the loop.
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly or view it on GitHub
> <https://github.com/wicketstuff/core/pull/494/files/f5e3c625ca3b47857f79d323fa64ed989b82c34a#r62912504>
>